src/test/java/pl/mjedynak/idea/plugins/builder/finder/BestConstructorSelectorTest.java

```diff
@@ -0,0 +1,154 @@
+    @Mock private PsiClass psiClass;
+    @Mock private PsiMethod constructor0;
+    @Mock private PsiMethod constructor1;
+    @Mock private PsiMethod constructor2a;
+    @Mock private PsiMethod constructor2b;
+    @Mock private PsiMethod constructor3;
+    @Mock private PsiParameterList parameterList0;
+    @Mock private PsiParameterList parameterList1;
+    @Mock private PsiParameterList parameterList2a;
+    @Mock private PsiParameterList parameterList2b;
+    @Mock private PsiParameterList parameterList3;
+    @Mock private PsiField psiField1;
+    @Mock private PsiField psiField2;
+    @Mock private PsiField psiField3;
+    @Mock private PsiField psiField4;
+    @Mock private PsiField psiField5;
+    @Mock private PsiParameter psiParameter1;
+    @Mock private PsiParameter psiParameter2;
+    @Mock private PsiParameter psiParameter3;
+    @Mock private PsiParameter psiParameter4;
+
+    private PsiFieldVerifier verifier = new PsiFieldVerifier();
+    private BestConstructorSelector finder = new BestConstructorSelector(verifier);
+
+    @Before
+    public void initMock() {
+        setField(verifier, "codeStyleSettings", settings);
+        given(settings.getParameterNamePrefix()).willReturn(EMPTY);
+        given(settings.getFieldNamePrefix()).willReturn(EMPTY);
+
+        mockPsiVariable(psiField1, NAME_1, PsiType.INT);
+        mockPsiVariable(psiField2, NAME_2, PsiType.INT);
+        mockPsiVariable(psiField3, NAME_3, PsiType.INT);
+        mockPsiVariable(psiField4, NAME_4, PsiType.INT);
+        mockPsiVariable(psiField5, NAME_5, PsiType.INT);
+
+        mockPsiVariable(psiParameter1, NAME_1, PsiType.INT);
+        mockPsiVariable(psiParameter2, NAME_2, PsiType.INT);
+        mockPsiVariable(psiParameter3, NAME_3, PsiType.INT);
+        mockPsiVariable(psiParameter4, NAME_4, PsiType.INT);
+
+        mockContructor(constructor0, parameterList0, EMPTY_PSI_PARAMETERS);
+        mockContructor(constructor1, parameterList1, psiParameter4);
+        mockContructor(constructor2a, parameterList2a, psiParameter1, psiParameter2);
+        mockContructor(constructor2b, parameterList2b, psiParameter1, psiParameter4);
+        mockContructor(constructor3, parameterList3, psiParameter1, psiParameter2, psiParameter3);
+    }
+
+    private void mockPsiVariable(PsiVariable psiVariable, String name, PsiPrimitiveType type) {
+        given(psiVariable.getName()).willReturn(name);
+        given(psiVariable.getType()).willReturn(type);
+    }
+
+    private void mockContructor(PsiParameterListOwner constructor, PsiParameterList parameterList, PsiParameter... psiParameters) {
+        given(constructor.getParameterList()).willReturn(parameterList);
+        given(parameterList.getParameters()).willReturn(psiParameters);
+        given(parameterList.getParametersCount()).willReturn(psiParameters.length);
+    }
+
+    @Test
+    public void shouldFindConstructorWithLeastParametersIfAnyFieldsToFind() {
+        doTest(

```

Comment:
seems like use case for https://github.com/Pragmatists/JUnitParams

Reply:
Thx for merging.

I tried JUnitParams, but using it with Mockito in this context is a hassle. Actually, JUnitParams calls the method collecting the params before @ Before method containing MockitoAnnotations.initMocks(this), so that mocks contained in the params are not initialized => NPE. This is documented here https://stackoverflow.com/questions/36218495/how-to-use-mockito-stubs-in-a-junitparams-parameter-provider/36219944

I achieved to work around it by moving the initialization of a part of the mocks inside the params method, and the other part in the test method itself. As this solution is more dirty than my first version, I prefer to leave it as it is now.

Perhaps I'm not aware of how I should do, and this can be sorted out. Let me know if you know more about it.